packages/components/src/border-control/border-control/component.tsx

```diff
@@ -14,16 +18,16 @@ import { useBorderControl } from './hook';
 import type { BorderControlProps, LabelProps } from '../types';
 
 const BorderLabel = ( props: LabelProps ) => {
-	const { label, hideLabelFromVision } = props;
+	const { className, label, hideLabelFromVision } = props;
 
 	if ( ! label ) {
 		return null;
 	}
 
 	return hideLabelFromVision ? (
-		<VisuallyHidden as="label">{ label }</VisuallyHidden>
+		<VisuallyHidden as="legend">{ label }</VisuallyHidden>
 	) : (
-		<StyledLabel>{ label }</StyledLabel>
+		<legend className={ className }>{ label }</legend>
```

Comment:
Could we simplify this by doing `<StyledLabel as="legend">`?

Also, I noticed that in Chrome/FF/Safari the user agent stylesheet adds 2px side padding to `legend` elements. I think we might be using the `<StyledLabel as="legend">` pattern in other places too, so I wonder if we should add the `padding: 0` to the [original styles](https://github.com/WordPress/gutenberg/blob/c777886d29c77da45f003e081a2691fecf2f37c2/packages/components/src/base-control/styles/base-control-styles.ts#L35) (with a comment).

Reply:
Thanks! That does help reduce the changes. 👍 

- Switched back to `StyledLabel` in https://github.com/WordPress/gutenberg/pull/42348/commits/ed78d2a775ec8c2a0b1606f3d5e097327344af1f
- Removed padding in base control's `StyleLabel` styles in https://github.com/WordPress/gutenberg/pull/42348/commits/d6f1c268a040103d803a07b48b5cc0ebe09bb396